sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/TableFileCatalog.scala

```diff
@@ -102,6 +95,13 @@ class TableFileCatalog(
   }
 
   override def inputFiles: Array[String] = allPartitions.inputFiles
+
+  override def equals(o: Any): Boolean = o match {
```

Comment:
Under hive context, we will cache the `LogicalRelation` for every data source table(including converted from hive), which means every table will always have a `TableFileCatalog` of same instance.

However, it's not true in sql core. We will re-construct the `TableFileCatalog` and `LogicalRelation` everytime we look up a table. Thus we may encounter cache miss even if the table is cached, because `TableFileCatalog` of difference instances never equal to each other.

Although it's not a real problem now, I think it's reasonable to follow `ListFileCatalong` and add the `equals` and `hashCode`